### DISCUSSION TITLE
Android lifecycle listeners and sending events to javascript

### DISCUSSION THREAD

**Discussion by 7ammer** [expo/expo#24741] [Archived: Questions]
Hello 👋 

Wondering if anyone can help.

I'm trying to create a module (https://docs.expo.dev/modules/get-started/) that listens for onConfigurationChanged and sends the locale value to javascript land.
My goal is something similar to this: https://medium.com/call-stack/react-native-handling-language-changes-on-android-the-right-way-c883056a8f5c

I've found out about android lifecycle listeners (https://docs.expo.dev/modules/android-lifecycle-listeners/) but I'm not sure how to send an event inside onConfigurationChanged 🤔.

Can anyone help me?

**Comment by SalmanK81099** [comment]
Hi @7ammer , did you find any solution for this? I am also working on the same, I am confused on how I can pass these listeners to the javascript file.

**Comment by adumat** [comment]
Watching the expo modules source code, there are no expo official way to handle this.
But, expo-localization to that task implementing a observer pattern.

You can see more about in the source code [Package][1] and [Module][2] of expo-localization


  [1]: https://github.com/expo/expo/blob/main/packages/expo-localization/android/src/main/java/expo/modules/localization/LocalizationPackage.kt
  [2]: https://github.com/expo/expo/blob/main/packages/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt
